Sources/CardPayments/APIRequests/ConfirmPaymentSourceRequest.swift

```diff
@@ -74,8 +75,80 @@ struct ConfirmPaymentSourceRequest: APIRequest {
         let cancelUrl: String
     }
     
+    enum CodingKeys: String, CodingKey {
+        case number
+        case expiry
+        case securityCode
+        case name
+        case billingAddress
+        case attributes
+    }
+    
+    private struct EncodedCard: Encodable {
+        
+        let number: String
+        let securityCode: String
+        let billingAddress: Address?
+        let name: String?
+        let expiry: String
+        let attributes: Attributes?
+                
+        func encode(to encoder: Encoder) throws {
+            var container = encoder.container(keyedBy: CodingKeys.self)
+            try container.encode(number, forKey: .number)
+            try container.encode(expiry, forKey: .expiry)
+            try container.encode(securityCode, forKey: .securityCode)
+            try container.encode(name, forKey: .name)
+            try container.encode(billingAddress, forKey: .billingAddress)
+            try container.encode(attributes, forKey: .attributes)
+        }
+    }
+    
+    private struct Verification: Encodable {
+        
+        let method: String
+    }
+    
+    private  struct CardVault: Encodable {
+        
+        let storeInVault: String?
+    }
+    
+    private struct Customer: Encodable {
+        
+        let id: String
+    }
+    
+    private struct Attributes: Encodable {
+        
+        var customer: Customer?
+        var vault: CardVault?
+        let verification: Verification
+        
+        init(vault: Vault? = nil, verificationMethod: String) {
+            self.verification = Verification(method: verificationMethod)
+            if let vault {
+                self.vault = CardVault(storeInVault: "ON_SUCCESS")
+                if let id = vault.customerID {
+                    self.customer = Customer(id: id)
+                }
+            }
+        }
+    }
+    
     private struct PaymentSource: Encodable {
         
-        let card: Card
+        var card: EncodedCard
```

Comment:
Take it or leave it, calling the property the same name may make this more clear - it can also be a constant
```suggestion
        let encodedCard: EncodedCard
```

Reply:
Yes, I was just looking at that. You are right.

Reply:
I appreciate all of your comments and feedback!